Move some window functions from WebProcessProxy to WebPageProxy

charliewolfe opened the pull request:
#### 0adc9c766fb1391a789d379cfbf86bc13a1a3d19
<pre>
Move some window functions from WebProcessProxy to WebPageProxy
<a href="https://bugs.webkit.org/show_bug.cgi?id=265608">https://bugs.webkit.org/show_bug.cgi?id=265608</a>
<a href="https://rdar.apple.com/119003812">rdar://119003812</a>

Reviewed by Alex Christensen.

To resolve &apos;FIXME: This, postMessageToRemote, renderTreeAsText, etc. should be messages to the
WebPageProxy instead of the process. They are more the page doing things than the process.&apos;

* Source/WebKit/UIProcess/RemotePageProxy.h:
(WebKit::RemotePageProxy::sendSync):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::sendToWebPage):
(WebKit::WebPageProxy::sendSyncToProcessContainingFrame):
I added helper functions to support sending synchronous IPC messages to a process containing a
specified frame.

(WebKit::WebPageProxy::closeRemoteFrame):
(WebKit::WebPageProxy::focusRemoteFrame):
(WebKit::WebPageProxy::postMessageToRemote):
(WebKit::WebPageProxy::renderTreeAsText):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::postMessageToRemote): Deleted.
(WebKit::WebProcessProxy::closeRemoteFrame): Deleted.
(WebKit::WebProcessProxy::focusRemoteFrame): Deleted.
(WebKit::WebProcessProxy::renderTreeAsText): Deleted.
* Source/WebKit/UIProcess/WebProcessProxy.h:
* Source/WebKit/UIProcess/WebProcessProxy.messages.in:
* Source/WebKit/WebProcess/WebCoreSupport/WebRemoteFrameClient.cpp:
(WebKit::WebRemoteFrameClient::postMessageToRemote):
(WebKit::WebRemoteFrameClient::renderTreeAsText):
(WebKit::WebRemoteFrameClient::close):
(WebKit::WebRemoteFrameClient::focus):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::didReceiveSyncMessage):
(WebKit::WebPage::remotePostMessage):
(WebKit::WebPage::renderTreeAsText):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::didReceiveSyncMessage):
(WebKit::WebProcess::remotePostMessage): Deleted.
(WebKit::WebProcess::renderTreeAsText): Deleted.
* Source/WebKit/WebProcess/WebProcess.h:
This change removed the remaining synchronous IPC messages going to `WebProcess`. As a result, the
`didReceiveSyncWebProcessMessage()` function declaration was no longer generated.

* Source/WebKit/WebProcess/WebProcess.messages.in:

Canonical link: <a href="https://commits.webkit.org/271401@main">https://commits.webkit.org/271401@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8dd35373a40118679a446909fe77886a3d0a0134

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/28191 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/6832 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/29544 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/30720 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/25685 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/8861 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/4215 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/25968 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/28459 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/5594 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/24259 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/4859 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/4995 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/31409 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/25801 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/25686 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/31309 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/4979 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/3156 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/29065 "Found 1 new API test failure: /WebKitGTK/TestWebKitWebView:/webkit/WebKitWebView/notification (failure)") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/6527 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/5416 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3652 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/5494 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -13276,6 +13274,51 @@ void WebPageProxy::sendToProcessContainingFrame(std::optional<FrameIdentifier> f
+void WebPageProxy::renderTreeAsText(WebCore::FrameIdentifier frameID, size_t baseIndent, OptionSet<WebCore::RenderAsTextFlag> behavior, CompletionHandler<void(String&&)>&& completionHandler)
+{
+    auto sendResult = sendSyncToProcessContainingFrame(frameID, Messages::WebPage::RenderTreeAsText(frameID, baseIndent, behavior));
+    auto [result] = sendResult.takeReply();
```
You have to check if sendResult.succeeded() before taking its reply.  Otherwise it'll crash sometimes.  We need to design this in a less error prone way.

charliewolfe replied:
Whoops, didn't mean to drop that check, will fix.

charliewolfe committed:
Move some window functions from WebProcessProxy to WebPageProxy
https://bugs.webkit.org/show_bug.cgi?id=265608
rdar://119003812

Reviewed by Alex Christensen.

To resolve 'FIXME: This, postMessageToRemote, renderTreeAsText, etc. should be messages to the
WebPageProxy instead of the process. They are more the page doing things than the process.'

* Source/WebKit/UIProcess/RemotePageProxy.h:
(WebKit::RemotePageProxy::sendSync):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::sendToWebPage):
(WebKit::WebPageProxy::sendSyncToProcessContainingFrame):
I added helper functions to support sending synchronous IPC messages to a process containing a
specified frame.

(WebKit::WebPageProxy::closeRemoteFrame):
(WebKit::WebPageProxy::focusRemoteFrame):
(WebKit::WebPageProxy::postMessageToRemote):
(WebKit::WebPageProxy::renderTreeAsText):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::postMessageToRemote): Deleted.
(WebKit::WebProcessProxy::closeRemoteFrame): Deleted.
(WebKit::WebProcessProxy::focusRemoteFrame): Deleted.
(WebKit::WebProcessProxy::renderTreeAsText): Deleted.
* Source/WebKit/UIProcess/WebProcessProxy.h:
* Source/WebKit/UIProcess/WebProcessProxy.messages.in:
* Source/WebKit/WebProcess/WebCoreSupport/WebRemoteFrameClient.cpp:
(WebKit::WebRemoteFrameClient::postMessageToRemote):
(WebKit::WebRemoteFrameClient::renderTreeAsText):
(WebKit::WebRemoteFrameClient::close):
(WebKit::WebRemoteFrameClient::focus):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::didReceiveSyncMessage):
(WebKit::WebPage::remotePostMessage):
(WebKit::WebPage::renderTreeAsText):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::didReceiveSyncMessage):
(WebKit::WebProcess::remotePostMessage): Deleted.
(WebKit::WebProcess::renderTreeAsText): Deleted.
* Source/WebKit/WebProcess/WebProcess.h:
This change removed the remaining synchronous IPC messages going to `WebProcess`. As a result, the
`didReceiveSyncWebProcessMessage()` function declaration was no longer generated.

* Source/WebKit/WebProcess/WebProcess.messages.in:

Canonical link: https://commits.webkit.org/271401@main
